spec/fixtures/generate_external_certs.sh

```diff
@@ -0,0 +1,322 @@
+testdomain.com
+EOM
+IP_ALTNAMES_VELUM=10.17.1.0
+
+# Read multiline block into variable
+read -r -d '' ALTNAMES_KUBEAPI_LN << EOM
+kubernetes
+kubernetes.default
+kubernetes.default.svc
+kubernetes.default.svc.cluster.local
+api
+api.infra.caasp.local
+kube-api-x1.devenv.caasp.suse.net
+master-0
+master-0.infra.caasp.local
+bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
+bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb.infra.caasp.local
+EOM
+IP_ALTNAMES_KUBEAPI=172.24.0.1
+
+# Read multiline block into variable
+read -r -d '' ALTNAMES_DEX_LN << EOM
+dex
+dex.kube-system
+dex.kube-system.svc
+dex.kube-system.svc.infra.caasp.local
+dex.kube-system.svc.cluster.local
+kubernetes
+kubernetes.default
+kubernetes.default.svc
+api
+api.infra.caasp.local
+kube-api-x1.devenv.caasp.suse.net
+master-0
+master-0.infra.caasp.local
+bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
+bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb.infra.caasp.local
+EOM
+IP_ALTNAMES_DEX=172.24.0.1
+
+ALTNAMES_VELUM=$(echo "$ALTNAMES_VELUM_LN" | tr "\n" -)
+ALTNAMES_KUBEAPI=$(echo "$ALTNAMES_KUBEAPI_LN" | tr "\n" -)
+ALTNAMES_DEX=$(echo "$ALTNAMES_DEX_LN" | tr "\n" -)
+
+# Strip extra dashes at end of line
+ALTNAMES_VELUM="${ALTNAMES_VELUM%?}"
+ALTNAMES_KUBEAPI="${ALTNAMES_KUBEAPI%?}"
+ALTNAMES_DEX="${ALTNAMES_DEX%?}"
+
+SITE_SUBJ="/ST=WA/O=Test/OU=Test Unit/CN=test.com"
+ALTNAMES_BAD=test.com-blah.com
+IP_ALTNAMES_BAD=127.0.0.1-13::17
+
+init
+init_root_ca    root                 "/CN=US/O=US/OU=US Unit"
+init_sub_ca     intermed    root     "/CN=US2/O=US2/OU=US2 Unit"
+init_sub_ca     intermed2   intermed "/CN=US3/O=US3/OU=US3 Unit"
+create_site_crt velum_a     intermed2 sha256 2048 good    "/CN=va$SITE_SUBJ"      $ALTNAMES_VELUM   $IP_ALTNAMES_VELUM
+create_site_crt velum_b     intermed2 sha256 2048 good    "/CN=vb$SITE_SUBJ"      $ALTNAMES_VELUM   $IP_ALTNAMES_VELUM
+create_site_crt kubeapi_a   intermed2 sha256 2048 good    "/CN=ka$SITE_SUBJ"      $ALTNAMES_KUBEAPI $IP_ALTNAMES_KUBEAPI
+create_site_crt kubeapi_b   intermed2 sha256 2048 good    "/CN=kb$SITE_SUBJ"      $ALTNAMES_KUBEAPI $IP_ALTNAMES_KUBEAPI
+create_site_crt dex_a       intermed2 sha256 2048 good    "/CN=da$SITE_SUBJ"      $ALTNAMES_DEX     $IP_ALTNAMES_DEX
+create_site_crt dex_b       intermed2 sha256 2048 good    "/CN=db$SITE_SUBJ"      $ALTNAMES_DEX     $IP_ALTNAMES_DEX 
+create_site_crt weak        intermed2 sha256 1024 good    "/CN=weak$SITE_SUBJ"    $ALTNAMES_VELUM   $IP_ALTNAMES_VELUM
+create_site_crt sha1_digest intermed2 sha1   2048 good    "/CN=sha1$SITE_SUBJ"    $ALTNAMES_VELUM   $IP_ALTNAMES_VELUM
+create_site_crt expired     intermed2 sha256 2048 expired "/CN=expired$SITE_SUBJ" $ALTNAMES_VELUM   $IP_ALTNAMES_VELUM
+create_site_crt badalt      intermed2 sha256 2048 good    "/CN=badalt$SITE_SUBJ"  $ALTNAMES_BAD     $IP_ALTNAMES_BAD
+create_site_crt noalt       intermed2 sha256 2048 good    "/CN=noalt$SITE_SUBJ"   0                 0
+
+for cert in $DIR_CERTS/ca_*.crt; do
+  mv "$cert" "external_certs/$(basename "$cert")"
```

Comment:
These might benefit from a `mv -v` for log output, so one can easily see what's being put in place?

Reply:
Added verbose output.